src/Open3D/GUI/Window.cpp

```diff
@@ -0,0 +1,1191 @@
+        ImGui::PushStyleVar(ImGuiStyleVar_WindowBorderSize,
+                            theme.dialogBorderWidth);
+        ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding,
+                            theme.dialogBorderRadius);
+        if (DrawChild(dc, "dialog", impl_->activeDialog, DIALOG) !=
+            Widget::DrawResult::NONE) {
+            needsRedraw = true;
+        }
+        ImGui::PopStyleVar(2);
+    }
+
+    // Finish frame and generate the commands
+    ImGui::PopFont();
+    ImGui::EndFrame();
+    ImGui::Render();  // creates the draw data (i.e. Render()s to data)
+
+    // Draw the ImGui commands
+    impl_->imgui.imguiBridge->update(ImGui::GetDrawData());
+
+    // Draw. Since ImGUI is an immediate mode gui, it does layout during
+    // draw, and if we are drawing for layout purposes, don't actually
+    // draw, because we are just going to draw again after this returns.
+    if (!isLayoutPass) {
+        impl_->renderer->BeginFrame();
+        impl_->renderer->Draw();
+        impl_->renderer->EndFrame();
+    }
+
+    if (needsLayout) {
+        return Widget::DrawResult::RELAYOUT;
+    } else if (needsRedraw) {
+        return Widget::DrawResult::REDRAW;
+    } else {
+        return Widget::DrawResult::NONE;
+    }
+}
+
+Window::DrawResult Window::OnDraw() {
+    bool neededLayout = impl_->needsLayout;
+
+    auto result = DrawOnce(neededLayout);
+    if (result == Widget::DrawResult::RELAYOUT) {
+        impl_->needsLayout = true;
+    }
+
+    // ImGUI can take two frames to do its layout, so if we did a layout
+    // redraw a second time. This helps prevent a brief red flash when the
+    // window first appears, as well as corrupted images if the
+    // window initially appears underneath the mouse.
+    if (neededLayout || impl_->needsLayout) {
+        DrawOnce(false);
+    }
+
+    return (result == Widget::DrawResult::NONE ? NONE : REDRAW);
+}
+
+void Window::OnResize() {
+    impl_->needsLayout = true;
+
+#if __APPLE__
+    // We need to recreate the swap chain after resizing a window on macOS
+    // otherwise things look very wrong.
+    impl_->renderer->UpdateSwapChain();
+#endif  // __APPLE__
+
+    impl_->imgui.imguiBridge->onWindowResized(*this);
+
+    auto size = GetSize();
+    auto scaling = GetScaling();
+
+    auto oldContext = MakeCurrent();
+    ImGuiIO& io = ImGui::GetIO();
+    io.DisplaySize = ImVec2(size.width, size.height);
+    if (impl_->imgui.scaling != scaling) {
+        updateImGuiForScaling(1.0 / impl_->imgui.scaling);  // undo previous
+        updateImGuiForScaling(scaling);
+        impl_->imgui.scaling = scaling;
+    }
+    io.DisplayFramebufferScale.x = 1.0f;
+    io.DisplayFramebufferScale.y = 1.0f;
+
+    if (impl_->wantsAutoSizeAndCenter) {
+        impl_->wantsAutoSizeAndCenter = false;
+        auto* monitor = glfwGetWindowMonitor(impl_->window);
+        if (monitor) {
+            const GLFWvidmode* mode = glfwGetVideoMode(monitor);
+            if (mode) {
```

Comment:
Yes, we should proceed and do something "reasonable". Definitely not supposed to die or throw; just because GLFW can't figure out what monitor the window is on doesn't mean we should abort. Actually, monitor seems to be null on macOS, which is rather annoying.